### ISSUE TITLE
Promote cosign to public API (and support bzlmod)

### ISSUE THREAD

**Issue by seh** [bazel-contrib/rules_oci#235]
In #36 we introduced support for using the _cosign_ tool with the `cosign_attest` and `cosign_sign` rules. Both of those require a registered toolchain for _cosign_. The _cosign/repositories.bzl_ file defines the `cosign_register_toolchains` macro, but it remains difficult to use in a _WORKSPACE.bzlmod_ file. Ideally we would register a toolchain for _cosign_ in the _MODULE.bazel_ file like we do for _crane_.

In short, it would be good to be able to use _cosign_ as easily as we can the other tools integrated by this module.

**Comment by alexeagle** [comment]
Yup, cosign is a "private API" in 1.0. Only the distroless team is using it. Adding bzlmod support is just one of the tasks for considering it "ready" to commit to support in the public API surface.

**Comment by farcop** [comment]
We want to use `cosign_attest` rule to, but unable to generate SBOM bazel way.
I have created https://github.com/chainguard-dev/rules_apko/issues/35

**Comment by seh** [comment]
Coming back to this desire, has there been any movement on exposing _cosign_ that isn't mentioned here?

**Comment by thesayyn** [comment]
Not really, i am not sure if cosign belongs here anymore. Only reason for keeping it is distroless, i thought about moving it to rules_distroless.

**Comment by farcop** [comment]
We would really like to use this, but there are two problems:
1. https://github.com/sigstore/cosign/pull/3371 it was fixed a time ago but not released yet
2. This issue.

**Comment by seh** [comment]
> Only reason for keeping it is distroless, i thought about moving it to rules_distroless.

You're far more familiar with the domain than I am, but this surprised me, as it's not clear to me what _cosign_ has to do with _distroless_. I'm looking to sign my container images, whether or not the images are based on _distroless_.

If not here, then a dedicated _rules_cosign_ ruleset would make sense, though I expect that it would rather small, paying the outsized tax of needing its own CI workflow, BCR integration, and releases.

**Comment by dsupru** [comment]
Need this as well
We use `oci_image` to make images, would make sense from a user's point of view to also sign it

Does it have to be a separate rule ?
(from a user's point of view) Would be nice to just have something like:
```
oci_image(
    name = "image",
    ...
    cosign_sign = "<define how to sign it here: path to the private key, etc..>"
)
```

**Comment by alexeagle** [comment]
If anyone has time to contribute this or can track down some OSS funding from their organization, that would be great. Primary maintainers here currently don't have volunteer time for it.

**Comment by salrashid123** [comment]
hi-  @alexeagle short of a direct contribuion, do you happen to  cite a sample of how to include `cosign_sign` locally?

what i mean by that is to use `rules_oci` to build stuff and push but since i can't cant' reference cosign_sign off of that, i'd setup just a local reference which did the same.   i don't know that much bazel rules/templates to write that from scratch so any pointers or end-to-end example would help (i think `distroless` one or two other repos did this but untangling just the cosing stuff gets pretty involved).

the other approach which i'd rather not do is invoke `sh_binary` and `run_binary` on cosign..
